src/transformers/models/opt/modeling_opt.py

```diff
@@ -64,18 +64,20 @@
 ]
 
 
-def _make_causal_mask(input_ids_shape: torch.Size, dtype: torch.dtype, past_key_values_length: int = 0):
+def _make_causal_mask(
+    input_ids_shape: torch.Size, dtype: torch.dtype, device: torch.device, past_key_values_length: int = 0
+):
     """
     Make causal mask used for bi-directional self-attention.
     """
     bsz, tgt_len = input_ids_shape
-    mask = torch.full((tgt_len, tgt_len), torch.tensor(torch.finfo(dtype).min))
-    mask_cond = torch.arange(mask.size(-1))
+    mask = torch.full((tgt_len, tgt_len), torch.tensor(torch.finfo(dtype).min, device=device), device=device)
+    mask_cond = torch.arange(mask.size(-1), device=device)
```

Comment:
```suggestion
    mask = torch.full((tgt_len, tgt_len), torch.finfo(dtype).min, device=device)
    mask_cond = torch.arange(mask.size(-1), device=device)
```
It seems that `torch.tensor(torch.finfo(torch.float32).min, device="cuda")` requires CPU-GPU sync

Reply:
Otherwise if a tensor is needed, we can also do
```
torch.ones(1, device="cuda") * torch.finfo(torch.float32).min # no sync
```

Reply:
I just tried my suggestion:`mask = torch.full((tgt_len, tgt_len), torch.finfo(dtype).min, device=device)` 
and it does work on "cpu". I'm not sure why it wouldn't work? 🤔

Reply:
From the pytorch doc (https://pytorch.org/docs/stable/generated/torch.full.html), `torch.full` allows for scalar type for `fill_value`. So I think what @NouamaneTazi is trying to convey is that you don't need to put it first cast `-inf` to a tensor type and then fill, you could just fill with `-inf`